tajo-core/src/test/java/org/apache/tajo/engine/query/TestJoinQuery.java

```diff
@@ -89,9 +89,9 @@ public TestJoinQuery(String joinOption) {
   @Parameters
   public static Collection<Object[]> generateParameters() {
     return Arrays.asList(new Object[][]{
-        {"Hash_NoBroadcast"},
```

Comment:
These parameterized test cases should be included.
Each query with different parameter returns different order of rows. Adding additional order by column or order by clause can resolve this problem as the following example:
testComplexJoinCondition7.sql
```
select
  n1.n_nationkey,
  n1.n_name,
  n2.n_name
from nation n1 join (select * from nation union select * from nation) n2 on substr(n1.n_name, 1, 4) = substr(n2.n_name, 1, 4)
order by n1.n_nationkey;
``` 

You can add "n2.n_name" column in where clause.

```
select
  n1.n_nationkey,
  n1.n_name,
  n2.n_name
from nation n1 join (select * from nation union select * from nation) n2 on substr(n1.n_name, 1, 4) = substr(n2.n_name, 1, 4)
order by n1.n_nationkey, n2.n_name;
```